src/org/labkey/test/tests/DomainDesignerTest.java

```diff
@@ -1498,6 +1507,172 @@ public void addUpdateRemoveConditionalFormat() throws Exception
+                            .setMvEnabled(true)
+                            .setConditionalFormats(List.of(new ConditionalFormat(List.of(eqFive), true, true, false))),
+                        new FieldDefinition("dateTimeField", FieldDefinition.ColumnType.DateAndTime)
+                            .setValidators(List.of(soonValidator))));
+        dgen.createList(createDefaultConnection(), "Key");
+        DomainDesignerPage domainDesignerPage = DomainDesignerPage.beginAt(this, getProjectName(), "lists", listName);
+        DomainFormPanel domainFormPanel = domainDesignerPage.fieldsPanel();
+
+        File exportFile = domainFormPanel.clickExportFields();
+        List<PropertyDescriptor> createdFields = getFieldsFromExportFile(exportFile);
+
+        PropertyDescriptor stringField = createdFields.stream().filter(a-> a.getName().equals("textField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected string field was not exported", stringField);
+        assertThat("description did not export", stringField.getDescription(), is("is texty"));
+        assertThat("label did not export", stringField.getLabel(), is("text field"));
+
+        PropertyDescriptor boolField = createdFields.stream().filter(a-> a.getName().equals("booleanField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected boolean field was not exported", boolField);
+        assertTrue(boolField.getRequired());
+
+        PropertyDescriptor intfield = createdFields.stream().filter(a-> a.getName().equals("intField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected integer field was not exported", intfield);
+        assertTrue("expect mvEnabled to be true", intfield.getMvEnabled());
+        assertEquals(1, intfield.getConditionalFormats().size());
+        assertEquals("format.column~eq=5", intfield.getConditionalFormats().get(0).queryFilterToJSONString());
+
+        PropertyDescriptor decimalField = createdFields.stream().filter(a-> a.getName().equals("decimalField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected decimal field was not exported", decimalField);
+        assertThat(decimalField.getFormat(), is("0.####"));
+
+        PropertyDescriptor attachmentField = createdFields.stream().filter(a-> a.getName().equals("attachmentField"))
+                .findFirst().orElse(null);
+        assertNotNull("expected attachment field was not exported", attachmentField);
+        assertTrue(attachmentField.getHidden());
+
+        for (PropertyDescriptor intendedField : dgen.getColumns())
+        {
+            PropertyDescriptor exportedField = createdFields.stream().filter(a-> a.getName().equals(intendedField.getName()))
+                    .findFirst().orElse(null);
+            assertNotNull("expected field ["+intendedField.getName()+"] was not exported", exportedField);
+            verifyExpectedFieldProperties(intendedField, exportedField);
```

Comment:
I like this "iterate through the dgen.getColumns() and compare with the exportedField" approach. I'm curious why then, the above checks exist on a per field basis? From what I can see the verifyExpectedFieldProperties covers all of the things that those checks before the for loop are doing. Am I missing something?

Reply:
I wrote those first, then added the iterative/parameterized one after and forgot to remove them